feat(entity): implement removeAttachment(permission)

Silthus opened the pull request:
# Description
Permissible EntityMock now implements removeAttachment to remove permissions from a mock player.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

Silthus committed:
feat(entity): implement removeAttachment(permission)

Permissible EntityMock now implements removeAttachment to remove permissions from a mock player.

seeseemelk committed:
Merge branch 'v1.16' into v1.16

seeseemelk committed:
gradle: bump version number

seeseemelk commented on src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:
```diff
@@ -376,9 +377,22 @@ public PermissionAttachment addAttachment(Plugin plugin, int ticks)
 	{
-		// TODO Auto-generated constructor stub
-		throw new UnimplementedOperationException();
+		if (attachment == null) {
+			throw new IllegalArgumentException("Attachment cannot be null");
+		}
+
+		if (permissionAttachments.contains(attachment)) {
+			permissionAttachments.remove(attachment);
+			PermissionRemovedExecutor ex = attachment.getRemovalCallback();
 
+			if (ex != null) {
+				ex.attachmentRemoved(attachment);
+			}
+
+			recalculatePermissions();
+		} else {
+			throw new IllegalArgumentException("Given attachment is not part of Permissible object " + this);
+		}
```
Please use Allman braces (opening and closing brace always one a separate line).

seeseemelk reviewed (approved):
Looks good, except for a small style note! Thanks for the PR!

Silthus committed:
style: fix allman braces style

Silthus commented:
Ok I fixed the style.

On a small side node: are you aware of [semantic-release](https://github.com/semantic-release/semantic-release). It could help you in keeping the version numbers up-to-date and remove the manual version bumping task.

You can see an example here: https://github.com/Silthus/spigot-plugin-template/blob/master/package.json

Silthus committed:
feat(server): implement getHelpMap()

Add the already implemented HelpMapMock to the ServerMock and return it.

Silthus committed:
fix(commands): get command from command map

dispatchCommand now searches the CommandMap instead of PluginCommands only.
This allows the execution of dynamically registered commands.

Silthus commented:
Oh damn, I did not see that this PR is still open and added another fix and feature for commands. Hope this is ok.

Here are the changes:

- fixed an issue where dynamically registered commands could not be executed.
- added the `getHelpMap()` implementation

seeseemelk committed:
Merge branch 'v1.16' into v1.16
